priam/src/main/java/com/netflix/priam/aws/SDBInstanceData.java

```diff
@@ -64,14 +67,18 @@
     }
 
     private final AWSCredentialsProvider provider;
+    private final ICredential crossAccountProvider;
     private final Region sdbRegion;
     private final String sdbDomain;
+    private final boolean sdbAssumeRole;
 
     @Inject
-    public SDBInstanceData(AWSCredentialsProvider provider, AmazonConfiguration amazonConfiguration) {
+    public SDBInstanceData(AWSCredentialsProvider provider, @Named("awsec2roleassumption")ICredential crossAccountProvider,  AmazonConfiguration amazonConfiguration, CassandraConfiguration cassandraConfiguration) {
```

Comment:
Don't see the need for injecting both of these.  I think it might be cleaner to inject a named credential provider, like `@SDBCredentialProvider AWSCredentialsProvider`.  Also, I think you can avoid the custom STS assume role provider and `ICredential` entirely by using a `STSAssumeRoleSessionCredentialsProvider`.